File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int zeros=0;
        int product = 1;
        for(int num:nums){
            if (num==0){
                zeros++;
            }else{
                product*=num;
            }
        }
        int[] res = new int[nums.length];
        for(int i=0;i<nums.length;i++){
            if(nums[i]!=0){
                if (zeros>0){
                    res[i]=0;
                }else{
                    res[i]=product/nums[i];
                }
            }else{
                if(zeros>1){
                    res[i]=0;
                }else{
                    res[i]=product;
                }
            }
        }
        return res;
    }
}  

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        
        def solve(ind, stk, csum):
            if csum==target:
                if stk not in res:
                    res.append(stk.copy())
                return
            if ind==len(nums) or csum>target:
                return
            stk.append(nums[ind])
            solve(ind,stk,csum+nums[ind])
            stk.pop()
            solve(ind+1, stk, csum)
        
        solve(0,[],0)

        return res